onnxruntime/core/platform/env.h

```diff
@@ -64,13 +65,28 @@ struct ThreadOptions {
   // the main thread, which is usually set in the main executable(not controlled by onnxruntime.dll).
   unsigned int stack_size = 0;
 
-  // Thread affinity means a thread can only run on the logical processors that the thread is allowed to run on.
-  // If the vector is not empty, set the affinity of each thread to just one CPU.
-  // Index is thread index, value is CPU ID, starting from zero. For example, the first thread in the pool will be bound
-  // to the logical processor with id of affinity[0]. If the vector is empty, the thread can run on all the processors
-  // its process can run on. NOTE: When hyperthreading is enabled, for example, on a 4 cores 8 physical threads CPU,
-  // processor group [0,1,2,3] may only contain half of the physical cores.
-  std::vector<size_t> affinity;
+  // Thread affinity means a thread can only run on the logical processor(s) that the thread is allowed to run on.
+  // If the vector is not empty, then set the affinity of each thread to logical cpus ids within the ThreadAffinity.
+  // For example, the first thread in the pool will be bound to the logical processors contained in affinity[0].
+  // If the vector is empty, the thread can run on all the processors its process can run on. 
+  // NOTE: When hyperthreading is enabled, for example, on a 4 cores 8 physical threads CPU,
+  // processor group [0,1,2,3] may only specify half of the physical cores. There might be more than 2 logical
+  // processor per physical core on a given computer. Physical cores assigned to a given VM may contain
+  // logical processor indices that do not start with 0.
+  // If the size of the TP is not specified, ORT creates thread pools with a number of threads that are equal
+  // to the number of visible physical cores. The threads affinities are set to all of the logical processors
+  // that are contained in a physical core with the same index as the TP thread. ORT does not set any affinity
+  // to the thread that is considered main (outside thread-pools).
+  // The process that owns the thread may consider setting its affinity.
+
+  /// <summary>
+  /// A set of logical processors per-thread
+  /// </summary>
+  struct ThreadAffinity {
+    std::vector<int> logical_proc_ids;
```

Comment:
>std::vector<int> logical_proc_ids; [](http://example.com/codeflow?start=4&length=34)

nit - maybe could simplify this by:
"using LogicalProcessors = std::vector<int>;"
further, 'proc' is not very good shorthand simply because to many linux folks this means process, not processor.